cmd/podman/rm.go

```diff
@@ -52,148 +48,30 @@ func init() {
 	markFlagHiddenForRemoteClient("latest", flags)
 }
 
-func joinContainerOrCreateRootlessUserNS(runtime *libpod.Runtime, ctr *libpod.Container) (bool, int, error) {
-	if os.Geteuid() == 0 {
-		return false, 0, nil
-	}
-	s, err := ctr.State()
-	if err != nil {
-		return false, -1, err
-	}
-	opts := rootless.Opts{
-		Argument: ctr.ID(),
-	}
-	if s == libpod.ContainerStateRunning || s == libpod.ContainerStatePaused {
-		data, err := ioutil.ReadFile(ctr.Config().ConmonPidFile)
-		if err != nil {
-			return false, -1, errors.Wrapf(err, "cannot read conmon PID file %q", ctr.Config().ConmonPidFile)
-		}
-		conmonPid, err := strconv.Atoi(string(data))
-		if err != nil {
-			return false, -1, errors.Wrapf(err, "cannot parse PID %q", data)
-		}
-		return rootless.JoinDirectUserAndMountNSWithOpts(uint(conmonPid), &opts)
+// rmCmd removes one or more containers
+func rmCmd(c *cliconfig.RmValues) error {
+	if c.Bool("trace") {
+		span, _ := opentracing.StartSpanFromContext(Ctx, "rmCmd")
+		defer span.Finish()
 	}
-	return rootless.BecomeRootInUserNSWithOpts(&opts)
-}
 
-// saveCmd saves the image to either docker-archive or oci
-func rmCmd(c *cliconfig.RmValues) error {
-	var (
-		deleteFuncs []shared.ParallelWorkerInput
-	)
 	if os.Geteuid() != 0 {
 		rootless.SetSkipStorageSetup(true)
 	}
 
-	ctx := getContext()
-	runtime, err := libpodruntime.GetRuntime(&c.PodmanCommand)
+	runtime, err := adapter.GetRuntime(&c.PodmanCommand)
 	if err != nil {
 		return errors.Wrapf(err, "could not get runtime")
 	}
 	defer runtime.Shutdown(false)
 
-	if rootless.IsRootless() {
```

Comment:
Did we lose all the rootless handling here? That's going to break a lot of things.

Reply:
it looks like he's just moved it to be dealt with where he gets the containers right @jwhonce

Reply:
But that doesn't work? The rootless logic is different for different commands - I think @giuseppe just overhauled the way `rm` joins namespaces to fix some issues.

@giuseppe can you confirm that?

Reply:
The presentation level can no longer do this kind of work as it's run on both the client and server.

I moved it into shortcuts.GetContainerByContext().  I'll be happy to work with @giuseppe if that is wrong.  But his last commit won't work with podman-remote.